src/toolkit/matl_buy_policy.cc

```diff
@@ -55,34 +53,45 @@ void MatlBuyPolicy::set_throughput(double x) {
   throughput_ = x;
 }
 
-void MatlBuyPolicy::set_active(int x) {
-  assert(x > 0);
-  active_ = x;
+void MatlBuyPolicy::set_default_distributions() {
+  active_dist_ = new FixedIntDist(1);
+  dormant_dist_ = new FixedIntDist(-1);
+  size_dist_ = new FixedDoubleDist(1.0);
+  init_active_dormant();
 }
 
-void MatlBuyPolicy::set_dormant(int x) {
-  assert(x >= 0);
-  dormant_ = x;
+void MatlBuyPolicy::init_active_dormant() {
+  next_active_end_ = active_dist_->sample();
```

Comment:
Doesn't this need to be added to `manager()->context()->time()` in case this policy gets initialized after time 0?

Or just call `SetNextActiveTime()`

Reply:
What happens if `next_active_end_ < 0`?

Reply:
If I add a call to `manager()->context()->time()` here I get a seg fault when building

Reply:
It's odd that it seg faults when building???  I don't have the fine details of the order of initializing everything in my head, but conceptually, I could imagine a seg fault if an agent invokes this initialization before the context has initialized time - I'd have to dig to see if that's what's happening and then figure out what the best path forward would be.  I think the agent has to exist (?? and been added to the context ??) before you `Init()` the buy policy.

In general, it seems that the first time that `next_active_end_` is set does need to be relative to the time at which the agent enters the simulation.  Calling `SetNextActiveTime()` should take care of that (but it may also result in a segfault?)

Our tests may not catch this and may always have agents enter at time 0 - if that's true, it's not your problem to fix as part of this PR, but we should make an issue to improve testing.

Reply:
👍 added check of dist max in init, throw ValueError if >1. Removed check in SampleRequestSize